### PR TITLE
Adds support for Rails 4 and drop support for Rails 2.3.

### PR DESCRIPTION
**Pull request by arturopie**
ActiveRecord::Observer was extracted from Rails 4 to its own gem. See (http://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-3-2-to-rails-4-0), so LazyObservers needs to require the files from that gem if running in Rails 4.

### CODE REVIEW

**Comment by arturopie** [comment]
oops, forgot to update Travis. Doing it now.

**Comment by arturopie** [comment]
Can I drop support for Ruby 1.8.7?

File: lib/lazy_observers/rails_observer.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+if ActiveRecord::VERSION::MAJOR < 4
+  require 'active_record/observer'
+else
+  begin
+    require 'rails/observers/activerecord/active_record'
+  rescue LoadError => _
+    $stderr.puts(<<-MSG)
```
Comment (grosser) [inline lib/lazy_observers/rails_observer.rb:7]: Hmm I like the message, but maybe raise instead of warn ?

**Comment by grosser** [comment]
kick that 1.8 out :D

**Comment by arturopie** [comment]
Done :D

**Comment by grosser** [comment]
so ... green :D

**Comment by grosser** [comment]
do you want to use raise or keep the warn ?

File: lib/lazy_observers/rails_observer.rb (hunk 1 above)
Comment (arturopie) [inline lib/lazy_observers/rails_observer.rb:7]: The next line will raise the original exception. That way the user get all the info from the original exception. So yes, it is still raising.

**Comment by arturopie** [comment]
It does both, it prints the error, and re-raises the original exception

**Comment by arturopie** [comment]
More than a warning, it is actually suggesting the user how to fix that `LoadError` exception.

**Comment by grosser** [comment]
perfect, did not see that :)

**Comment by grosser** [comment]
FYI rails 4.0.x does not get security updates, so if you have an app on that you might want to update to 4.1/4.2

**Comment by grosser** [comment]
0.2.0 is out ! :)

**Comment by arturopie** [comment]
Thanks! Yes, I'm going to upgrade my app to 4.2. 4.0 is just an intermediate step.

**Pull request merged**